### PR TITLE
book: use `cargo add` when adding dependencies

### PR DESCRIPTION
**Pull request by kianmeng**

### CODE REVIEW

File: book/src/libadwaita.md [hunk 1]
```diff
@@ -11,10 +11,10 @@ Libadwaita is a library augmenting GTK 4 which:
 - allows runtime recoloring with [named colors](https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/named-colors.html)
 - adds [API](https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita/struct.StyleManager.html) to support the cross-desktop dark style preference
 
-In order to use the Rust bindings add the following line to your dependencies in `Cargo.toml`, where `X.X` should be replaced with the most up-to-date version of the [libadwaita crate](https://crates.io/crates/libadwaita).
+In order to use the Rust bindings, add the [libadwaita crate](https://crates.io/crates/libadwaita) as dependency to `Cargo.toml` using following command:
 
-```toml
-adw = { version = "X.X", package = "libadwaita" }
+```
+cargo add libadwaita
```
Comment (Hofer-Julian) [inline book/src/libadwaita.md:17]: Libadwaita needs to be renamed to adw.

cargo add supports that

File: book/src/project_setup.md [hunk 2]
```diff
@@ -9,10 +9,10 @@ Now, create a new project by executing:
 cargo new my-gtk-app
 ```
 
-Add the following line to your dependencies in `Cargo.toml`, where `X.X` should be replaced with the most up-to-date version of the [gtk4 crate](https://crates.io/crates/gtk4).
+Add the [gtk4 crate](https://crates.io/crates/gtk4) to your dependencies in `Cargo.toml` using following command:
 
-```toml
-gtk = { version = "X.X", package = "gtk4" }
+```
+cargo add gtk4
```
Comment (Hofer-Julian) [inline book/src/project_setup.md:15]: Rename to gtk

File: book/src/project_setup.md [hunk 3]
```diff
@@ -22,6 +22,13 @@ gtk = { version = "X.X", package = "gtk4" }
 > ```toml
 >gtk = { version = "X.X", package = "gtk4", features = ["v4_6"]}
 >```
+>
+> Or using the following command:
+>
+> ```
```
Comment (Hofer-Julian) [inline book/src/project_setup.md:28]: Rename to gtk

**Review by Hofer-Julian — CHANGES_REQUESTED** [review]
Thanks.

I‘ve added minor comments.

i also think there are more places in the book where dependencies are introduced.

File: book/src/libadwaita.md (hunk 1 above)
Comment (kianmeng) [reply book/src/libadwaita.md:17]: If I understand you correctly? Can't find `adw`?

```sh
$ cargo add adw
    Updating crates.io index
error: the crate `adw` could not be found in registry index.
```

File: book/src/project_setup.md (hunk 2 above)
Comment (kianmeng) [reply book/src/project_setup.md:15]: It returned GTK3?

```sh
$ cargo add gtk
    Updating crates.io index
      Adding gtk v0.15.5 to dependencies.
             Features:
             - dox
             - gio_v2_58
             - v3_20
             - v3_22
             - v3_22_20
             - v3_22_26
             - v3_22_27
             - v3_22_29
             - v3_22_30
             - v3_24
             - v3_24_1
             - v3_24_11
             - v3_24_30
             - v3_24_8
             - v3_24_9
```

```toml
[dependencies]
gtk = "0.15.5"
```

**Comment by kianmeng** [comment]
> i also think there are more places in the book where dependencies are introduced.

Noted, and updated.

File: book/src/libadwaita.md (hunk 1 above)
Comment (Hofer-Julian) [reply book/src/libadwaita.md:17]: cargo add libadwaita --rename=adw

File: book/src/todo_2.md [hunk 4]
```diff
@@ -196,6 +196,11 @@ Combined with [`serde_json`](https://lib.rs/crates/serde_json) we can save our t
 
 First, we extend our `Cargo.toml` with the `serde` and `serde_json` crate.
 
+```
+cargo add serde --derive
```
Comment (Hofer-Julian) [inline book/src/todo_2.md:200]: That doesn't look right.

Shouldn't that be something like `--feature=derive`?

Comment (kianmeng) [reply book/src/todo_2.md]: You're right. Fixed.

**Comment by Hofer-Julian** [comment]
Thanks!

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

Thanks for your kind patience for guiding me through this.

**Pull request merged**